src/index.ts

```diff
@@ -0,0 +1,168 @@
+import * as _ from 'underscore';
+import { AWSProvider, ServerlessInstance, ServerlessConfig } from './types';
+
+interface PluginConfig {
+  stages?: string[];
+  roleArn?: string;
+  filterPattern: string;
+  normalizedFilterID: boolean;
+  createLambdaPermission: boolean;
+  destinationARN: string;
+}
+
+const configDefaults = {
+  filterPattern: '',
+  normalizedFilterID: true,
+  createLambdaPermission: true,
+};
+
+interface CFObject<TProps> {
+  Type: string;
+  DependsOn?: string[];
+  Properties: TProps;
+}
+
+interface LambdaPermissionProps {
+  Action: string;
+  Principal: string;
+  FunctionName: string;
+}
+
+interface SubscriptionFilterProps {
+  DestinationArn: string;
+  FilterPattern: string;
+  LogGroupName: string;
+  RoleArn?: string;
+}
+
+type LambdaPermission = CFObject<LambdaPermissionProps>;
+type SubscriptionFilter = CFObject<SubscriptionFilterProps>;
+
+function getStage(options: ServerlessConfig, sls: ServerlessInstance) {
+  if (options.stage && options.stage !== '') {
+    return options.stage;
+  }
+  return sls.service.provider.stage;
+}
+
+class LogForwardingPlugin {
+  options: ServerlessConfig;
+
+  provider: AWSProvider;
+
+  serverless: ServerlessInstance;
+
+  config: PluginConfig | null = null;
+
+  hooks: object;
+
+  constructor(serverless: ServerlessInstance, options: ServerlessConfig) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+    this.hooks = {
+      'package:initialize': this.updateResources.bind(this),
+    };
+  }
+
+  loadConfig() {
+    this.config = { ...configDefaults, ...this.serverless.service.custom.logForwarding };
```

Comment:
You need to add a validation here for example check `this.serverless.service.custom` option is specified

Reply:
addressed

Reply:
addressed